src/shared/redux/slices/publicationsSlice.ts

```diff
@@ -34,24 +61,73 @@ const publicationsSlice = createSlice({
   name: 'publications',
   initialState,
   reducers: {
+    setPublicationsList: (
+      state: PublicationList,
+      action: PayloadAction<PublicationList>
+    ) => {
+      state = action.payload;
+    },
     addPublication: (
       state: PublicationList,
-      action: PayloadAction<PublicationListElement>
+      action: PayloadAction<Publication>
     ) => {
       state.list.push(action.payload);
     },
-    setPublicationList: (
+    deletePublication: (
+      state: PublicationList,
+      action: PayloadAction<string>
+    ) => {
+      const updatedPublications = state.list.filter(
+        (publication) => publication.id !== action.payload
+      );
+      state.list = updatedPublications;
+    },
+    setPublicationField: (
+      state: PublicationList,
+      action: PayloadAction<PublicationModification>
+    ) => {
+      const chosenPubIndex = state.list.findIndex(
+        (publication) => publication.id === action.payload.id
+      );
+      state.list[chosenPubIndex][action.payload.field] = action.payload.value;
+    },
+    addCollaborator: (
+      state: PublicationList,
+      action: PayloadAction<CollaboratorListModification>
+    ) => {
+      const chosenPubIndex = state.list.findIndex(
+        (publication) => publication.id === action.payload.pubId
+      );
+      const collaborator = action.payload.value as Collaborator;
+      state.list[chosenPubIndex].collaborators.push(collaborator);
+    },
+    deleteCollaborator: (
       state: PublicationList,
-      action: PayloadAction<[PublicationListElement]>
+      action: PayloadAction<CollaboratorListModification>
     ) => {
-      state.list = action.payload;
+      const chosenPubIndex = state.list.findIndex(
+        (publication) => publication.id === action.payload.pubId
+      );
```

Comment:
Can we standarize the `action.payload.id` / `action.payload.pubId` in the actions so the naming convention is the same?